### ISSUE TITLE
workspace config dir continuously being polled

### ISSUE THREAD

**Issue by dromer** [microsoft/vscode-cpptools#11806]
### Environment

- OS and Version: Debian 12
- VS Code Version: 1.85.1
- C/C++ Extension Version: 1.18.5

### Bug Summary and Steps to Reproduce

Bug Summary:
Extension seems to cause continuous polling of the workspace config directory (`.vscode/` on Linux)

Steps to reproduce:
1. Have extension enabled
2. run `sudo strace -f -t -e trace=file -p <vscode_pid>`
3. see disk access

Expected behavior:
To not have extraneous access to this directory.

### Configuration and Logs
After doing an extension troubleshoot from the vscode help (bisecting the loaded extensions) this extension ended up being the culprit of the extensive disk access.

For now I have this extension completely disabled.

**Comment by bobbrow** [comment]
You are correct.  We poll this location because there have been cases where VS Code's file watchers do not notify us of configuration changes in the project which caused confusion for developers.  Because it is critical for us to always know the correct configuration we check the timestamp of the configuration file on an interval timer.  If this is a problem for you we could consider making this configurable.

**Comment by dromer** [comment]
The problem is the extreme amount of polling that is done. Maybe making this indeed more configurable (like every second/minute/hour/off or whatever) so the filesystem isn't being bombarded with requests.

I only noticed this behavior rather recently, so I don't know if there were any changes made to this.
